src/mesh/mesh_triangle_holes.C

```diff
@@ -159,14 +227,227 @@ Point TriangulatorInterface::ArbitraryHole::point(const unsigned int n) const
+
+              for (auto b : bcids)
+                {
+                  if (ids.empty() || ids.count(b))
+                    {
+                      hole_edge_map.emplace(elem->node_ptr(s),
+                                            elem->node_ptr((s+1)%ns));
+                      hole_edge_map.emplace(elem->node_ptr((s+1)%ns),
+                                            elem->node_ptr(s));
+                      break;
+                    }
+                }
+            }
+        }
+    }
+
+  libmesh_error_msg_if
+    (hole_edge_map.empty(),
+     "No valid hole edges found in mesh!");
+
+  std::vector<const Node *> hole_points
+    {hole_edge_map.begin()->first, hole_edge_map.begin()->second};
+
+  // Sort the edges into a connected order
+  for (const Node * last = hole_points.front(),
+                  *    n = hole_points.back();
+                       n != hole_points.front();
+                    last = n,
+                       n = hole_points.back())
+    {
+      auto [next_it_begin, next_it_end] = hole_edge_map.equal_range(n);
+
+      libmesh_error_msg_if
+        (std::distance(next_it_begin, next_it_end) != 2,
+         "Bad edge topology in MeshedHole");
+
+      const Node * next = nullptr;
+      for (auto [key, val] : as_range(next_it_begin, next_it_end))
+        {
+          libmesh_assert_equal_to(key, n);
+          libmesh_assert_not_equal_to(val, n);
+          if (val == last)
+            continue;
+          next = val;
+        }
+
+      hole_points.push_back(next);
+    }
+
+  hole_points.pop_back();
+
+  libmesh_error_msg_if
+    (hole_points.size() < 3,
+     "Only " << hole_points.size() << " hole edges found in mesh!");
+
+  _points.resize(hole_points.size());
+  std::transform(hole_points.begin(),
+                 hole_points.end(),
+                 _points.begin(),
+                 [](const Node * n){ return Point(*n); });
+
+  Real current_area = this->area();
+
+  libmesh_error_msg_if
+    (!current_area,
+     "Zero-area MeshedHoles are not currently supported");
+
+  // We ordered ourselves backwards?  Whoops, fix that.
+  if (current_area < 0)
+    std::reverse(_points.begin(), _points.end());
+
+  mesh.comm().broadcast(_points);
+}
+
+
+unsigned int TriangulatorInterface::MeshedHole::n_points() const
+{
+  return _points.size();
+}
+
+
+Point TriangulatorInterface::MeshedHole::point(const unsigned int n) const
+{
+  libmesh_assert_less (n, _points.size());
+  return _points[n];
+}
+
+
+Point TriangulatorInterface::MeshedHole::inside() const
+{
+  // This is expensive to compute, so only do it when we first need it
+  if (_center(0) == std::numeric_limits<Real>::max())
+    {
+      // Start with the vertex average
+
+      // Turns out "I'm a fully compliant C++17 compiler!" doesn't
+      // mean "I have a full C++17 standard library!"
+      // _center = std::reduce(_points.begin(), _points.end());
+      _center = 0;
+      for (auto p : _points)
```

Comment:
FYI, I tested and 
```
  std::vector<Point> v = {Point(0,0), Point(1,0), Point(1,1), Point(0,1)};
  Point result = std::accumulate(v.begin(), v.end(), Point());
```
compiles and does the right thing on my ancient compiler.

Reply:
You know, I got so wrapped up in the new exciting parallelizable algorithms that I completely forgot that one.  Good idea.